### PR TITLE
OI-255 Login page

### PR DESCRIPTION
**Pull request by nvelychenko**

### CODE REVIEW

File: config/install/editor.editor.comments.yml [hunk 1]
```diff
@@ -29,10 +29,12 @@ settings:
       status: ''
     mentions:
       enable: 1
+      mentions_type: user
```
Comment (db-Matroskeen) [inline config/install/editor.editor.comments.yml]: I'm just curious, how user mentions are relevant to the block on the login form?

File: config/install/rules.reaction.earn_points_for_voting_a_idea.yml [hunk 2]
```diff
@@ -22,8 +22,25 @@ expression:
     conditions:
       -
         id: rules_condition
-        uuid: e42d9597-8b68-4c0f-8104-82cf94775eed
+        uuid: 64e03746-3bac-4912-8821-0f309c585a53
```
Comment (db-Matroskeen) [inline config/install/rules.reaction.earn_points_for_voting_a_idea.yml]: Same here. How it's relevant to login form? :)

File: modules/openideal_user/openideal_user.install [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * Implements hook_update_N().
+ */
+function openideal_user_update_8001(&$sandbox) {
+  $blocks = \Drupal::entityTypeManager()->getStorage('block_content')->loadByProperties(['info' => 'login']);
+  if (!empty($blocks)) {
+    return;
+  }
+
+  // Create a block_content for right section of login screen.
+  $module_handler = \Drupal::service('module_handler');
```
Comment (db-Matroskeen) [inline modules/openideal_user/openideal_user.install]: So there is no way to import a single block from default_content content storage?

File: themes/openideal_theme/openideal_theme.theme [hunk 4]
```diff
@@ -68,16 +68,21 @@ function openideal_theme_preprocess_html(&$variables) {
...
+  $route_name = $route_match->getRouteName();
   $node = $route_match->getParameter('node');
   if ($node instanceof NodeInterface) {
     $variables['node_type'] = $node->bundle();
   }
 
-  if ($route_match->getRouteName() == 'view.frontpage.front_page'
-    || $route_match->getRouteName() == 'entity.user.canonical'
-    || $route_match->getRouteName() == 'view.news.all_news_page') {
+  if ($route_name == 'view.frontpage.front_page'
```
Comment (db-Matroskeen) [inline themes/openideal_theme/openideal_theme.theme]: I would prefer to replace multiple || with the following statement:
```
$some_routes = [];
in_array($route_name, $some_routes)
```

File: themes/openideal_theme/openideal_theme.theme [hunk 5]
```diff
@@ -68,16 +68,21 @@ function openideal_theme_preprocess_html(&$variables) {
...
+    || $route_name == 'entity.user.canonical'
+    || $route_name == 'view.news.all_news_page') {
     $variables['is_activity_page'] = TRUE;
   }
+  elseif ($route_name == 'user.login') {
```
Comment (db-Matroskeen) [inline themes/openideal_theme/openideal_theme.theme]: Why can't attach this library with block content?
If there is no block, then it's not necessary to attach the library, right?

**Review by db-Matroskeen — CHANGES_REQUESTED** [review]

File: themes/openideal_theme/openideal_theme.theme (hunk 4 above)
Comment (nvelychenko) [reply themes/openideal_theme/openideal_theme.theme]: Done

File: modules/openideal_user/openideal_user.install (hunk 3 above)
Comment (nvelychenko) [reply modules/openideal_user/openideal_user.install]: I moved the whole logic related to that block and user.login page into the openideal_login module, please check it out.

File: config/install/rules.reaction.earn_points_for_voting_a_idea.yml (hunk 2 above)
Comment (nvelychenko) [reply config/install/rules.reaction.earn_points_for_voting_a_idea.yml]: Removed

File: config/install/editor.editor.comments.yml (hunk 1 above)
Comment (nvelychenko) [reply config/install/editor.editor.comments.yml]: Removed

File: modules/openideal_login/openideal_login.module [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+  }
+
+  // Remove socials that aren't configured properly.
+  foreach ($social_plugins as $plugin_id => &$data) {
+    if (!$network_manager->createInstance($plugin_id)->getSdk()) {
+      unset($social_plugins[$plugin_id]);
+    }
+
+    $theme_path = \Drupal::service('theme_handler')->getTheme('openideal_theme')->getPath();
+    $prefix = 'Sign in with';
```
Comment (db-Matroskeen) [inline modules/openideal_login/openideal_login.module]: Should be wrapped with t().

File: modules/openideal_login/openideal_login.module [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+    'reset' => [
+      '#type' => 'html_tag',
+      '#attributes' => [
+        'class' => [
+          'user-login-form--container__bottom--forgot-password',
+          'user-login-form--container__bottom--text',
+        ],
+      ],
+      '#tag' => 'p',
+      '#value' => t("@link", ['@link' => $reset_link]),
```
Comment (db-Matroskeen) [inline modules/openideal_login/openideal_login.module]: What's the purpose of this t() function call?

File: themes/openideal_theme/templates/components/login-with.html.twig [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{% set module = 'social-auth' %}
+
+{{ attach_library('social_auth/auth-icons') }}
+
+{% for social_network in social_networks %}
+  {% if destination %}
+    <a class="{{ module }} auth-link" href="{{ path(social_network.route) }}?destination={{ destination }}">
```
Comment (db-Matroskeen) [inline themes/openideal_theme/templates/components/login-with.html.twig:7]: Can we prepare url with all necessary parameters outside of the twig template?

**Review by db-Matroskeen — CHANGES_REQUESTED** [review]

File: themes/openideal_theme/templates/components/login-with.html.twig (hunk 8 above)
Comment (nvelychenko) [reply themes/openideal_theme/templates/components/login-with.html.twig:7]: I reworked social-auth template. Don't want to change their theme variables and logic. But if it's necessary I can do this.

File: modules/openideal_login/openideal_login.module (hunk 6 above)
Comment (nvelychenko) [reply modules/openideal_login/openideal_login.module]: The variable was unneeded, I removed it and put the 'Sign in with' inside t() function.

File: modules/openideal_login/openideal_login.module (hunk 7 above)
Comment (nvelychenko) [reply modules/openideal_login/openideal_login.module]: Done..

File: themes/openideal_theme/templates/components/login-with.html.twig (hunk 8 above)
Comment (db-Matroskeen) [reply themes/openideal_theme/templates/components/login-with.html.twig:7]: Please ignore this request, I thought this is a custom template.

**Review by db-Matroskeen — APPROVED** [review]
Looks good to me.
@nvelychenko, please merge into 3.x-dev.

**Pull request merged**